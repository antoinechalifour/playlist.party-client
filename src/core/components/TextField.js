import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.div`
  display: block;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
`

const Inner = styled.div`
  position: relative;
  padding-top: 0.5rem;
  margin-top: 0.5rem;
  padding-bottom: 0.5rem;
  border-width: 1px;
  border-style: solid;
  border-color: ${({ isFocused }) => (isFocused ? 'rgba(0, 0, 0, 0.5)' : 'rgba(0, 0, 0, 0.25)')};
  border-radius: 4px;
  transition: all .2s ease;
`

const Label = styled.label`
  position: absolute;
  transition: all .2s ease;
  left: 8px;
  color: ${({ theme }) => theme.colors.typographyLight};

  ${({ isTop }) => (isTop ? `
    top: 0;
    transform: translateY(-50%);
    font-size: 0.9rem;
    background: #fff;
    padding: 0 12px;
  ` : `
    top: 25%;
    font-size: inherit;
    background: transparent;
    padding: 0;
  `)}
`

const Input = styled.input`
  box-sizing: border-box;
  width: 100%;
  border: none;
  background: transparent;
  font-size: 0.95rem;
  font-family: inherit;
  outline: none;
  padding: 0 8px;
`

export default class TextField extends Component {
  static propTypes = {
    className: PropTypes.string,
    type: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired
  }

  static defaultProps = {
    type: 'text'
  }

  state = {
    isFocused: false
  }

  render () {
    return (
      <Wrapper className={this.props.className}>
        <Inner isFocused={this.state.isFocused}>
          <Label
            htmlFor={this.props.name}
            isTop={this.state.isFocused || this.props.value !== ''}
          >
            {this.props.label}
          </Label>
          <Input
            type={this.props.type}
            name={this.props.name}
            id={this.props.name}
            value={this.props.value}
            placeholder={this.props.placeholder}
            onChange={e => this.props.onChange(e.target.value)}
            onFocus={() => this.setState({ isFocused: true })}
            onBlur={() => this.setState({ isFocused: false })}
          />
        </Inner>
      </Wrapper>
    )
  }
}
