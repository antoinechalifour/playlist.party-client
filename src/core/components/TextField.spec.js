import React from 'react'
import renderer from 'react-test-renderer'
import TextField from './TextField'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'

const theme = {
  colors: {
    dark: '#151515',
    danger: '#a63d40',
    primary: '#1db954',
    confirm: '#90a959',
    accent: '#6494aa',
    typography: '#373d3f',
    typographyLight: '#bcbcbc'
  }
}

describe('TextField', () => {
  it('Should render an empty text TextField by default', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <TextField
            className='test-textfield'
            label='My label'
            name='test'
            value=''
            placeholder='write something'
            onChange={jest.fn()}
          />
        </ThemeProvider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Should render an empty Textfield', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <TextField
            className='test-textfield'
            type='email'
            label='Email address'
            name='email'
            value=''
            placeholder='john.doe@company.com'
            onChange={jest.fn()}
          />
        </ThemeProvider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Should render a filled Textfield', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <TextField
            className='test-textfield'
            type='email'
            label='Email address'
            name='email'
            value='jane.doe@company.com'
            placeholder='john.doe@company.com'
            onChange={jest.fn()}
          />
        </ThemeProvider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
