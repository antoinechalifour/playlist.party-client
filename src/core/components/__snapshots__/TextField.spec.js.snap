// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField Should render a filled Textfield 1`] = `
.c0 {
  display: block;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.c1 {
  position: relative;
  padding-top: 0.5rem;
  margin-top: 0.5rem;
  padding-bottom: 0.5rem;
  border-width: 1px;
  border-style: solid;
  border-color: rgba(0,0,0,0.25);
  border-radius: 4px;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
}

.c2 {
  position: absolute;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  left: 8px;
  color: #bcbcbc;
  top: 0;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  font-size: 0.9rem;
  background: #fff;
  padding: 0 12px;
}

.c3 {
  box-sizing: border-box;
  width: 100%;
  border: none;
  background: transparent;
  font-size: 0.95rem;
  font-family: inherit;
  outline: none;
  padding: 0 8px;
}

<div
  className="test-textfield c0"
>
  <div
    className="c1"
  >
    <label
      className="c2"
      htmlFor="email"
    >
      Email address
    </label>
    <input
      className="c3"
      id="email"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="john.doe@company.com"
      type="email"
      value="jane.doe@company.com"
    />
  </div>
</div>
`;

exports[`TextField Should render an empty Textfield 1`] = `
.c0 {
  display: block;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.c1 {
  position: relative;
  padding-top: 0.5rem;
  margin-top: 0.5rem;
  padding-bottom: 0.5rem;
  border-width: 1px;
  border-style: solid;
  border-color: rgba(0,0,0,0.25);
  border-radius: 4px;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
}

.c2 {
  position: absolute;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  left: 8px;
  color: #bcbcbc;
  top: 25%;
  font-size: inherit;
  background: transparent;
  padding: 0;
}

.c3 {
  box-sizing: border-box;
  width: 100%;
  border: none;
  background: transparent;
  font-size: 0.95rem;
  font-family: inherit;
  outline: none;
  padding: 0 8px;
}

<div
  className="test-textfield c0"
>
  <div
    className="c1"
  >
    <label
      className="c2"
      htmlFor="email"
    >
      Email address
    </label>
    <input
      className="c3"
      id="email"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="john.doe@company.com"
      type="email"
      value=""
    />
  </div>
</div>
`;

exports[`TextField Should render an empty text TextField by default 1`] = `
.c0 {
  display: block;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.c1 {
  position: relative;
  padding-top: 0.5rem;
  margin-top: 0.5rem;
  padding-bottom: 0.5rem;
  border-width: 1px;
  border-style: solid;
  border-color: rgba(0,0,0,0.25);
  border-radius: 4px;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
}

.c2 {
  position: absolute;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  left: 8px;
  color: #bcbcbc;
  top: 25%;
  font-size: inherit;
  background: transparent;
  padding: 0;
}

.c3 {
  box-sizing: border-box;
  width: 100%;
  border: none;
  background: transparent;
  font-size: 0.95rem;
  font-family: inherit;
  outline: none;
  padding: 0 8px;
}

<div
  className="test-textfield c0"
>
  <div
    className="c1"
  >
    <label
      className="c2"
      htmlFor="test"
    >
      My label
    </label>
    <input
      className="c3"
      id="test"
      name="test"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="write something"
      type="text"
      value=""
    />
  </div>
</div>
`;
